flink-core/src/main/java/org/apache/flink/core/plugin/PluginLoader.java

```diff
@@ -177,16 +177,16 @@ public P next() {
                     return resolveIfNeeded(resolve, loadedClass);
                 }
 
-                if (isAllowedFlinkClass(name)) {
-                    try {
+                try {
```

Comment:
We probably should have a separate CL if it doesn't work by default. I wonder what the issue is.

The original version, goes to Flink CL for all `o.a.f` classes (=API). Now we first load from plugin. I'm assuming if a user screws up and bundles `flink-core` etc. (original motivation for parent.first list), all hell breaks loose. 

Do we have an issue here because akka RPC also resides in `o.a.f`? It might be enough to strip down the parent.first list though... If it doesn't contain `o.a.f`, the change probably is unneeded.

Reply:
If it doesnt contain o.a.f then it cant use stuff like Preconditions. Id also prefer a separate CL, or a generic one flexible enough to allow certain parent-first / child-first patterns while denying everything not listed in the pattern.

Reply:
Could you help me understand why it doesn't work in IDE?

Reply:
Maybe, I'm not entirely sure myself.

Symptom: a NoSuchMethodError on some call from Flink into scala

Observation: IntelliJ puts both flink-rpc-akka and dependencies (akka, scala) on the test classpath, and because of the plugin loader these are then loaded parent-first.

Theory: Because the flink-rpc-akka classes are loaded parent-first IntelliJ is not using what is in the flink-rpc-akka jar, but instead what it compiled itself. I think it mixes up the scala versions somehow, in that it puts scala 2.12 on the CP (which is correct) but compiled flink-rpc-akka with scala 2.11. I don't know how InteliiJ determines which Scala version to use for compilation; it could very well be an issue in my IDE configuration.
To reproduce the issue, revert the [experimental] commit and run the `RpcConnectionTest`

By forcing the plugin loader to go child-first we always load what is actually in the flink-rpc-akka jar, which is guaranteed to be consistent in regards to scala in itself.

Reply:
I haven't tried running the test with the scala 2.12 profile being enabled, because I can't enable it without IntelliJ shitting the bed.